src/grpc/infer_handler.cc

```diff
@@ -432,9 +422,17 @@ InferGRPCToInput(
                 .c_str());
       }
       void* tmp;
+      std::shared_ptr<const SharedMemoryManager::SharedMemoryInfo> shm_info =
+          nullptr;
       RETURN_IF_ERR(shm_manager->GetMemoryInfo(
-          region_name, offset, byte_size, &tmp, &memory_type, &memory_type_id));
+          region_name, offset, byte_size, &tmp, &memory_type, &memory_type_id,
+          &shm_info));
       base = tmp;
+
+      if (shm_info != nullptr) {
```

Comment:
Should be able to remove this, a successful call to `GetMemoryInfo` should return valid `shm_info`.